src/core/src/bootstrap/EnvLayer.py

```diff
@@ -284,24 +284,30 @@ def open(self, file_path, mode):
                     if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT - 1:
                         time.sleep(i + 1)
                     else:
-                        raise Exception("Unable to open {0} (retries exhausted). Error: {1}.".format(str(real_path), repr(error)))
+                        #todo: do not print this error trace when raise_if_not_found is False. Also do not retry if raise_if_not_found is False
+                        error_message = "Unable to open {0} (retries exhausted). Error: {1}.".format(str(real_path), repr(error))
+                        print(error_message)
+                        if raise_if_not_found:
+                            raise Exception(error_message)
+                        else:
+                            return None
```

Comment:
Please change error message to the following for diagnosability:
"Unable to open file (retries exhausted). [File={0}][Error={1}][RaiseIfNotFound={2}].".format(str(real_path), repr(error),str(raise_if_not_found))

Sanity check expected behavior in all callers that come through to here.

Reply:
@benank - to take in a faster PR, and deploy to Canary for validation. May need additional changes.

Reply:
Resolved